packages/rich-text/src/component/index.js

```diff
@@ -286,20 +286,6 @@ class RichText extends Component {
 
 		this.recalculateBoundaryStyle();
 
-		// We know for certain that on focus, the old selection is invalid. It
-		// will be recalculated on the next mouseup, keyup, or touchend event.
```

Comment:
In some cases there will be a focus event without any pointer or keyboard event, in which case the selection is still the same.